File: app/src/main/java/de/phash/manuel/mathetrainer/EinfachesRechnen.java
package de.phash.manuel.mathetrainer;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.Locale;
import java.util.Objects;

public class EinfachesRechnen extends AppCompatActivity {

    ArrayList<Button> buttons = new ArrayList<>(4);
    Integer aufgabe = 0;
    Integer richtig = 0;
    Integer versuche = 0;
    private MatheAufgabe aktAufgabe;
    private String rechenart;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_einfaches_rechnen);
        rechenart = Objects.requireNonNull(getIntent().getExtras()).get(Konstanten.RECHENART).toString();
        buttons.add((Button) findViewById(R.id.loesung1));
        buttons.add((Button) findViewById(R.id.loesung2));
        buttons.add((Button) findViewById(R.id.loesung3));
        buttons.add((Button) findViewById(R.id.loesung4));

        updateViews();
    }

    public void loesungButtonClick(View view) {
        Button b = (Button) view;
        Integer gedrueckt = Integer.valueOf(b.getText().toString());
        Log.i("button", gedrueckt + "");

        if (aktAufgabe.getRichtigeLoesung().equals(gedrueckt)) {
            richtig++;
            aufgabe++;
            versuche++;
            Toast.makeText(this, "Richtig!", Toast.LENGTH_SHORT).show();
            updateViews();
        } else {
            versuche++;
            b.setBackgroundColor(Color.RED);
            updateVersuche();
            Toast.makeText(this, "leider falsch", Toast.LENGTH_SHORT).show();

        }
    }

    private void updateVersuche() {
        TextView versucheView = (TextView) findViewById(R.id.versucheView);
        versucheView.setText(versuche.toString());
    }

    private void updateViews() {
        updateVersuche();
        aktAufgabe = erzeugeMatheAufgabe();
        TextView aufgabeTextView = (TextView) findViewById(R.id.aktuelleAufgabe);
        aufgabeTextView.setText(aktAufgabe.getAufgabe());

        TextView geloesteAufgabenTextView = (TextView) findViewById(R.id.geloesteAufgaben);
        geloesteAufgabenTextView.setText(String.format(Locale.GERMAN, "%d", richtig));


        int i = 0;
        for (Integer ele : aktAufgabe.getLoesungen()) {
            buttons.get(i).setBackgroundColor(Color.LTGRAY);
            buttons.get(i).setText(Integer.toString(ele));
            i++;
        }
    }

    private MatheAufgabe erzeugeMatheAufgabe() {
        if (rechenart.equals(Konstanten.ADD))
            return MatheAufgaben.getAddition(4, 20);
        if (rechenart.equals(Konstanten.SUB))
            return MatheAufgaben.getSubstraktion(4, 20);
        return null;
    }

    public void hauptMenue(View view) {
        Intent intent = new Intent(this, MainActivity.class);
        startActivity(intent);
    }
}

File: app/src/main/java/de/phash/manuel/mathetrainer/EinfachesRechnen.kt
package de.phash.manuel.mathetrainer

import android.content.Context
import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import java.util.*

class EinfachesRechnen : AppCompatActivity() {

    internal var buttons = ArrayList<Button>(4)

    private var aktAufgabe: MatheAufgabe? = null
    private var rechenart: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_einfaches_rechnen)
        rechenart = Objects.requireNonNull(intent.extras).get(Konstanten.RECHENART)!!.toString()
        buttons.add(findViewById(R.id.loesung1) as Button)
        buttons.add(findViewById(R.id.loesung2) as Button)
        buttons.add(findViewById(R.id.loesung3) as Button)
        buttons.add(findViewById(R.id.loesung4) as Button)

        updateViews()
    }

    fun loesungButtonClick(view: View) {
        val b = view as Button
        val gedrueckt = Integer.valueOf(b.text.toString())
        Log.i("button", gedrueckt.toString() + "")

        if (aktAufgabe!!.richtigeLoesung == gedrueckt) {
            increaseAllStates()
            Toast.makeText(this, "Richtig!", Toast.LENGTH_SHORT).show()
            updateViews()
        } else {
            val sharedPref = this.getSharedPreferences("de.phash.manuel.mathetrainer", Context.MODE_PRIVATE)

            var versuchPers = sharedPref.getInt(Status.VERSUCHE, 0)
            sharedPref.edit().putInt(Status.VERSUCHE, versuchPers++).apply()
            Status.instance.versuche++

            sharedPref.edit().putInt(Status.VERSUCHEALLTIME, (sharedPref.getInt(Status.VERSUCHEALLTIME, 0) + 1)).apply()
            b.setBackgroundColor(Color.RED)
            b.isEnabled = false
            updateVersuche()
            Toast.makeText(this, "leider falsch", Toast.LENGTH_SHORT).show()

        }
    }

    private fun increaseAllStates() {
        StatsUpdateService.instance.increaseAllStates(this.getSharedPreferences("de.phash.manuel.mathetrainer", Context.MODE_PRIVATE))
    }

    private fun updateVersuche() {
        val versucheView = findViewById(R.id.versucheView) as TextView
        versucheView.text = Status.instance.versuche.toString()
    }

    private fun updateViews() {
        updateVersuche()

        aktAufgabe = erzeugeMatheAufgabe(Status.instance.bisMax)

        val aufgabeTextView = findViewById(R.id.aktuelleAufgabe) as TextView
        aufgabeTextView.text = aktAufgabe!!.aufgabe

        val geloesteAufgabenTextView = findViewById(R.id.geloesteAufgaben) as TextView
        geloesteAufgabenTextView.text = String.format(Locale.GERMAN, "%d", Status.instance.richtig)


        var i = 0
        for (ele in aktAufgabe!!.loesungen) {
            buttons[i].setBackgroundColor(Color.LTGRAY)
            buttons[i].isEnabled = true
            buttons[i].text = String.format(Locale.GERMAN, Integer.toString(ele!!))
            i++
        }
    }

    private fun erzeugeMatheAufgabe(bisMax: Int): MatheAufgabe? {
        if (rechenart == Konstanten.ADD)
            return MatheAufgaben.getAddition(4, bisMax)
        return if (rechenart == Konstanten.SUB) MatheAufgaben.getSubstraktion(4, bisMax) else null
    }

    fun hauptMenue(view: View) {
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
    }
}

